api/manager.py

```diff
@@ -13,33 +13,198 @@
+        item = db.execute(query).fetchone()
+        print(item)
+
+        if item is None:
+            error = "Not implemented"
+        elif len(item == 0):
+            error = "Empty table"
+
+        if error:
+            return json.dumps({"error": error})
+        else:
+            return json.dumps({"item": item})
 
     if request.method == "PUT":
+        # grab data
+        data = request.get_json()
+        displayName = data["display_name"]
+        stock = data["stock"]
+        pricePerUnit = data["price_per_unit"]
+        productType = data["product_type"]
+
+        # initiating variable
         db = get_db()
+        error = None
+
+        # check data entries
+        if not displayName:
+            error = "Display name missing."
+        elif not stock:
+            error = "Stock missing."
+        elif not pricePerUnit:
+            error = "Price per unit missing."
+        elif not productType:
+            error = "Product type missing."
+
+        # continue if no errors
+        if error is None:
+            try:
+                query = """
+                    REPLACE INTO product (display_name, stock, price_per_unit, product_type)
+                    VALUES (?, ?, ?, ?)
+                    """
+                db.execute(query, (displayName, stock, pricePerUnit, productType))
+                db.commit()
+            except:
+                error = "Error saving values"
+            else:
+                return json.dumps({"success": f"{displayName} has been updated."})
+        else:
+            return json.dumps({"error": error})
         # update record on one specific product
 
     if request.method == "POST":
+        # grab data
+        data = request.get_json()
+        displayName = data["display_name"]
+        stock = data["stock"] if "stock" in data else 0
+        pricePerUnit = data["price_per_unit"]
+        productType = data["product_type"]
+
+        # initiating variable
         db = get_db()
-        # add a new product
+        error = None
+
+        # check data entries
+        if not displayName:
+            error = "Product name is required."
+        elif not pricePerUnit:
+            error = "Price per unti is required."
+        elif not productType:
+            error = "Product type is required."
+
+        # continue if no errors
+        if error is None:
+            try:
+                query = """
+                    INSERT INTO product (display_name, stock, price_per_unit, product_type)
+                    VALUES (?, ?, ?, ?)
+                    """
+                db.execute(query, (displayName, stock, pricePerUnit, productType))
+                db.commit()
+            except:
+                error = "Error saving product."
+            else:
+                return json.dumps({"success": f"{productType} added successfully."})
+        else:
+            return json.dumps({"error": error})
 
 
 @bp.route("/user", methods=["GET", "PUT"])
```

Comment:
gonna wanna add an id in this route in the same way that i mentioned above (my b for not doing this in the example, was still working out how to do setup) and since the original comment i remembered u also put the same id in the function parameters eg. "/user/<user_id>" and then def user(user_id)